src/Components/Web/ref/Microsoft.AspNetCore.Components.Web.netcoreapp.cs

```diff
@@ -71,7 +71,7 @@ public abstract partial class InputBase<TValue> : Microsoft.AspNetCore.Component
         [Microsoft.AspNetCore.Components.ParameterAttribute]
         public System.Linq.Expressions.Expression<System.Func<TValue>>? ValueExpression { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
         protected virtual void Dispose(bool disposing) { }
-        protected virtual string? FormatValueAsString(TValue value) { throw null; }
+        protected virtual string? FormatValueAsString([System.Diagnostics.CodeAnalysis.AllowNullAttribute] TValue value) { throw null; }
```

Comment:
Question for the benefit of my ignorance. Is the attribute needed when you've explicitly stated the type is a nullable string? I thought the compiler did that for us?

Reply:
The `AllowNull` is for the parameter, which has the generic type `TInput`. I believe you are referring to the `string?` return type, which indeed does not need a `MaybeNull` attribute. Let me know if I misunderstood your question.

Reply:
Sorry, I half focused on the wrong thing. What I should have asked is why use the attribute on TValue when you could use `TValue?` as you do for the return type?

Reply:
In this case, `TValue?` is not possible because there are no generic type constraints on `TValue`, so the compiler cannot tell whether to resolve to a `Nullable<TValue>` or a nullable reference type `TValue?`.